enginetest/join_planning_tests.go

```diff
@@ -391,14 +391,14 @@ order by 1;`,
 				q: `SELECT * FROM xy WHERE (
       				EXISTS (SELECT * FROM xy Alias1 WHERE Alias1.x = (xy.x + 1))
       				AND EXISTS (SELECT * FROM uv Alias2 WHERE Alias2.u = (xy.x + 2)));`,
-				types: []plan.JoinType{plan.JoinTypeSemiLookup, plan.JoinTypeSemiLookup},
+				types: []plan.JoinType{plan.JoinTypeSemiLookup},
 				exp:   []sql.Row{{0, 2}, {1, 0}},
 			},
 			{
 				q: `SELECT * FROM xy WHERE (
       				EXISTS (SELECT * FROM xy Alias1 WHERE Alias1.x = (xy.x + 1))
       				AND EXISTS (SELECT * FROM uv Alias1 WHERE Alias1.u = (xy.x + 2)));`,
-				types: []plan.JoinType{plan.JoinTypeSemiLookup, plan.JoinTypeSemiLookup},
```

Comment:
We should fix this before merging.

Reply:
Reverted this change.

Reply:
The reason these changes are happening, is because we're no longer hoisting the filters, and so these remain subqueries.

For this SQL:
```sql
SELECT * FROM xy WHERE (
      				EXISTS (SELECT * FROM xy Alias1 WHERE Alias1.x = (xy.x + 1))
      				AND EXISTS (SELECT * FROM uv Alias2 WHERE Alias2.u = (xy.x + 2)));
```

Here's the starting tree:
```txt
Project
 ├─ columns: [*]
 └─ Filter
     ├─ AND
     │   ├─ EXISTS Subquery
     │   │   ├─ cacheable: false
     │   │   └─ Project
     │   │       ├─ columns: [*]
     │   │       └─ Filter
     │   │           ├─ (Alias1.x = (xy.x + 1))
     │   │           └─ TableAlias(Alias1)
     │   │               └─ UnresolvedTable(xy)
     │   └─ EXISTS Subquery
     │       ├─ cacheable: false
     │       └─ Project
     │           ├─ columns: [*]
     │           └─ Filter
     │               ├─ (Alias2.u = (xy.x + 2))
     │               └─ TableAlias(Alias2)
     │                   └─ UnresolvedTable(uv)
     └─ Table
         ├─ name: xy
         └─ columns: [x y]
```

And the final tree:
```txt
QueryProcess
 └─ Filter
     ├─ AND
     │   ├─ EXISTS Subquery
     │   │   ├─ cacheable: false
     │   │   └─ Filter
     │   │       ├─ (Alias1.x = (xy.x + 1))
     │   │       └─ TableAlias(Alias1)
     │   │           └─ Table
     │   │               ├─ name: xy
     │   │               └─ columns: [x y]
     │   └─ EXISTS Subquery
     │       ├─ cacheable: false
     │       └─ Filter
     │           ├─ (Alias2.u = (xy.x + 2))
     │           └─ TableAlias(Alias2)
     │               └─ Table
     │                   ├─ name: uv
     │                   └─ columns: [u v]
     └─ Table
         ├─ name: xy
         └─ columns: [x y]
```